src/gatling/simulations/scenarios/Caseworker_PRL_FL401_ProgressCase.scala

```diff
@@ -0,0 +1,486 @@
+      .check(jsonPath("$.documents[0]._links.self.href").saveAs("DocumentURLSoN")))
+     }
+
+  /*======================================================================================
+  * Service of Application document uploads
+  ======================================================================================*/
+
+    .group("XUI_PRL_XXX_660_DocumentUpload") {
+      exec(http("XUI_PRL_660_005_DocumentUpload")
+        .post(BaseURL + "/data/case-types/PRLAPPS/validate?pageId=serviceOfApplication2")
+        .headers(Headers.xuiHeader)
+        .header("accept", "application/vnd.uk.gov.hmcts.ccd-data-store-api.case-data-validate.v2+json;charset=UTF-8")
+        .header("x-xsrf-token", "#{XSRFToken}")
+        .body(ElFileBody("bodies/prl/courtAdmin/PRLSoADocumentsFL401.json"))
+        .check(substring("additionalDocuments")))
+    }
+
+    .pause(MinThinkTime, MaxThinkTime)
+
+  /*======================================================================================
+  * Service of Application Confirm recipients
+  ======================================================================================*/
+
+    .group("XUI_PRL_XXX_670_ServiceRecipients") {
+      exec(http("XUI_PRL_XXX_670_005_ServiceRecipients")
+        .post(BaseURL + "/data/case-types/PRLAPPS/validate?pageId=serviceOfApplication3")
+        .headers(Headers.xuiHeader)
+        .header("accept", "application/vnd.uk.gov.hmcts.ccd-data-store-api.case-data-validate.v2+json;charset=UTF-8")
+        .header("x-xsrf-token", "#{XSRFToken}")
+        .body(ElFileBody("bodies/prl/courtAdmin/PRLSoARecipientsFL401.json"))
+        .check(substring("soaServingRespondentsOptionsDA")))
+    }
+
+    .pause(MinThinkTime, MaxThinkTime)
+
+  /*======================================================================================
+  * Service of Application Submit
+  ======================================================================================*/
+
+    .group("XUI_PRL_XXX_680_ServiceSubmit") {
+      exec(http("XUI_PRL_XXX_680_005_ServiceSubmit")
+        .post(BaseURL + "/data/cases/#{caseId}/events")
+        .headers(Headers.xuiHeader)
+        .header("accept", "application/vnd.uk.gov.hmcts.ccd-data-store-api.create-event.v2+json;charset=UTF-8")
+        .header("x-xsrf-token", "#{XSRFToken}")
+        .body(ElFileBody("bodies/prl/courtAdmin/PRLSoASubmitFL401.json"))
+        .check(jsonPath("$.data.caseInvites[0].value.accessCode").saveAs("prlAccessCodeApplicant"))
+        .check(jsonPath("$.data.caseInvites[1].value.accessCode").saveAs("prlAccessCodeRespondent")))
+    }
+
+    //Write applicant access code to file
+    .exec { session =>
+      val fw = new BufferedWriter(new FileWriter("FL401caseNumberAndCodeApplicant.csv", true))
+      try {
+        fw.write(session("caseId").as[String] + "," + session("prlAccessCodeApplicant").as[String] + "\r\n")
+      } finally fw.close()
+      session
+    }
+    //Write respondent access code to file
```

Comment:
This is duplicated?

Reply:
No this is for applicant & Respondent data